### PR TITLE
Fix timestamp calculation and account creation bugs

### LINKED ISSUE #5266: Incorrect "last generated at" timestamp
### Actual behavior
image

shows the current timestamp, not the timestamp of the last block generated by the validator

### Which version(s) does this affect? (Environment, OS, etc...)
desktop 3.0.0-beta.1

### PR DESCRIPTION
**Pull request by ManuGowda**
### What was the problem?

This PR resolves #5266, #5236

### How was it solved?

- Fix last forger timestamp calculation
- Fix account creation to not use legacy address derivation
- Bug fixes related to blockchain application and few other minor UI bugs.

### How was it tested?

- view validators tab and go over some validator profile and check the last forged timestamp
- Use latest protocol with CDP for account creation
- CSS and text changes for UI bugs

### CODE REVIEW

**Review by oskarleonard — DISMISSED** [review]

File: src/modules/pos/validator/components/ValidatorProfile/DetailsView.js [hunk 1]
```diff
@@ -31,10 +30,10 @@ const DetailsView = ({ data, isMyProfile }) => {
   const history = useHistory();
   const theme = useTheme();
   const { t } = useTranslation();
-  // @TODO: this is wrong we need to get this value from the validator's endpoint.
-  const {
-    data: { timestamp: latestBlockTimestamp },
-  } = useLatestBlock();
+  const { data: generateBlock } = useBlocks({
+    config: { params: { height: lastGeneratedHeight } },
+    options: { enable: !lastGeneratedHeight },
```
Comment (eniolam1000752) [inline src/modules/pos/validator/components/ValidatorProfile/DetailsView.js]: Shouldn't this be

```
!!lastGeneratedHeight
```

File: src/modules/pos/validator/components/ValidatorProfile/DetailsView.js [hunk 2]
```diff
@@ -31,10 +30,10 @@ const DetailsView = ({ data, isMyProfile }) => {
   const history = useHistory();
   const theme = useTheme();
   const { t } = useTranslation();
-  // @TODO: this is wrong we need to get this value from the validator's endpoint.
-  const {
-    data: { timestamp: latestBlockTimestamp },
-  } = useLatestBlock();
+  const { data: generateBlock } = useBlocks({
```
Comment (eniolam1000752) [inline src/modules/pos/validator/components/ValidatorProfile/DetailsView.js]:
```suggestion
  const { data: generatedBlock } = useBlocks({
```

File: src/modules/pos/validator/components/ValidatorProfile/DetailsView.js [hunk 3]
```diff
@@ -64,8 +63,12 @@ const DetailsView = ({ data, isMyProfile }) => {
     {
       icon: 'calendar',
       label: t('Last generated at'),
-      value: latestBlockTimestamp ? (
-        <DateTimeFromTimestamp fulltime className="date" time={latestBlockTimestamp} />
+      value: generateBlock?.data?.[0] ? (
```
Comment (eniolam1000752) [inline src/modules/pos/validator/components/ValidatorProfile/DetailsView.js]:
```suggestion
      value: generatedBlock?.data?.[0] ? (
```

File: src/modules/pos/validator/components/ValidatorProfile/DetailsView.js [hunk 4]
```diff
@@ -64,8 +63,12 @@ const DetailsView = ({ data, isMyProfile }) => {
...
+        <DateTimeFromTimestamp
+          fulltime
+          className="date"
+          time={generateBlock?.data?.[0].timestamp}
```
Comment (eniolam1000752) [inline src/modules/pos/validator/components/ValidatorProfile/DetailsView.js]:
```suggestion
          time={generatedBlock?.data?.[0].timestamp}
```

**Review by eniolam1000752 — CHANGES_REQUESTED** [review]

**Review by eniolam1000752 — DISMISSED** [review]

File: src/modules/pos/validator/components/ValidatorProfile/DetailsView.js [hunk 5]
```diff
@@ -31,10 +30,10 @@ const DetailsView = ({ data, isMyProfile }) => {
   const history = useHistory();
   const theme = useTheme();
   const { t } = useTranslation();
-  // @TODO: this is wrong we need to get this value from the validator's endpoint.
-  const {
-    data: { timestamp: latestBlockTimestamp },
-  } = useLatestBlock();
+  const { data: generatedBlock } = useBlocks({
+    config: { params: { height: lastGeneratedHeight } },
+    options: { enable: !!lastGeneratedHeight },
```
Comment (ikem-legend) [inline src/modules/pos/validator/components/ValidatorProfile/DetailsView.js]: Typo here. Should be `enabled`

**Review by ikem-legend — DISMISSED** [review]

**Review by oskarleonard — APPROVED** [review]

**Review by ikem-legend — APPROVED** [review]

**Review by eniolam1000752 — APPROVED** [review]

**Pull request merged**